### PR TITLE
Ensure co-located test files don't have definitions generated

### LINKED ISSUE #464: Typings for test files end up in dist folder
### Current Behavior

If you run 

```sh
npx tsdx create test-project # choose react
```
, create a test file (say `index.test.tsx`) with some content, and run `yarn build`, you'll end up with a file called `dist/index.test.d.ts`. 

### Expected behavior

TSDX shouldn't create this.

### Suggested solution(s)

Perhaps there's a filter you could apply? Idk 🤷‍♂ Haven't had time to research it yet.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | @latest
| TypeScript       | not installed
| Browser          | n/a
| npm/Yarn         | yarn 1x
| Node             |  lts
| Operating System | osx

### PR DESCRIPTION
**Pull request by selbekk**
This commit fixes an issue where type declaration files were created
for test files as well.

Fixes #464

### CODE REVIEW

**Comment by selbekk** [comment]
Updated the list of excludes in all templates. I can centralise in the defaults if you want - especially since the list got a bit longer?

**Comment by agilgur5** [comment]
I would think it's better to add these universally as it makes the excludes transparent for folks with project structures like yours, and makes no changes to those with `test/` directories either. In templates that means people with `test/` would have to delete the excludes (or leave them in redundantly); some might also not understand that they're not necessarily needed for all projects.

In any case, let's wait for some thoughts from the other maintainers as they might decline wholesale these changes, idk

**Comment by agilgur5** [comment]
@sw-yx any thoughts on adding this to templates vs. adding it to `tsconfigDefaults` (of rpts2)?

**Comment by sw-yx** [comment]
i dont have strong feelings about this, i can see arguments for both sides. happy to go with whatever you and @selbekk want to do.

**Comment by selbekk** [comment]
If we add it to the tsconfigDefaults, we can fix this issue for both existing users of tsdx and new ones.

**Comment by selbekk** [comment]
I'm not sure why this isn't working on the windows-latest build - should work as expected?

File: src/createRollupConfig.ts [hunk 1]
```diff
@@ -140,6 +140,16 @@ export async function createRollupConfig(
...
         tsconfigDefaults: {
+          exclude: [
+            '**/*.spec.ts',
+            '**/*.test.ts',
+            '**/*.spec.tsx',
+            '**/*.test.tsx',
+            'node_modules',
+            'bower_components',
+            'jspm_packages',
+            'dist',
```
Comment (agilgur5) [inline src/createRollupConfig.ts]:
```suggestion
            paths.appDist,
```

File: src/createRollupConfig.ts [hunk 2]
```diff
@@ -140,6 +140,16 @@ export async function createRollupConfig(
         cacheRoot: `./node_modules/.cache/tsdx/${opts.format}/`,
         tsconfig: opts.tsconfig,
         tsconfigDefaults: {
+          exclude: [
+            '**/*.spec.ts',
+            '**/*.test.ts',
+            '**/*.spec.tsx',
+            '**/*.test.tsx',
```
Comment (agilgur5) [inline src/createRollupConfig.ts]:
```suggestion
            '**/*.test.tsx',
            // TS defaults below
```

File: src/createRollupConfig.ts [hunk 3]
```diff
@@ -140,6 +140,16 @@ export async function createRollupConfig(
         cacheRoot: `./node_modules/.cache/tsdx/${opts.format}/`,
         tsconfig: opts.tsconfig,
         tsconfigDefaults: {
+          exclude: [
```
Comment (agilgur5) [inline src/createRollupConfig.ts]:
```suggestion
          exclude: [
          // all TS test files, regardless whether co-located or in test/ etc
```

**Comment by sw-yx** [comment]
@all-contributors pls thank @selbekk for code

**Comment by agilgur5** [comment]
@all-contributors please add @selbekk for code

**Pull request merged**